File: esports-backend/src/main/java/com/esports/util/ValidationUtil.java
package com.esports.util;

import java.util.regex.Pattern;

public class ValidationUtil {
    private static final Pattern EMAIL_PATTERN = Pattern.compile("^[A-Za-z0-9+_.-]+@(.+)$");
    private static final Pattern NAME_PATTERN = Pattern.compile("^[a-zA-Z\\s]{2,50}$");
    
    public static boolean isValidEmail(String email) {
        return email != null && EMAIL_PATTERN.matcher(email).matches();
    }
    
    public static boolean isValidName(String name) {
        return name != null && NAME_PATTERN.matcher(name.trim()).matches();
    }
    
    public static String sanitizeInput(String input) {
        if (input == null) return null;
        return input.trim().replaceAll("[<>\"'&]", "");
    }
}

File: esports-backend-dotnet/Upkart1/Utils/ValidationUtil.cs
using System.Text.RegularExpressions;

namespace Upkart1.Utils
{
    public static class ValidationUtil
    {
        private static readonly Regex EMAIL_PATTERN = new(@"^[A-Za-z0-9+_.-]+@(.+)$", RegexOptions.Compiled);
        private static readonly Regex NAME_PATTERN = new(@"^[a-zA-Z\s]{2,50}$", RegexOptions.Compiled);
        
        public static bool IsValidEmail(string? email)
        {
            return email != null && EMAIL_PATTERN.IsMatch(email);
        }
        
        public static bool IsValidName(string? name)
        {
            return name != null && NAME_PATTERN.IsMatch(name.Trim());
        }
        
        public static string? SanitizeInput(string? input)
        {
            if (input == null) return null;
            return Regex.Replace(input.Trim(), @"[<>""'&]", "");
        }
    }
}